src/main/java/Duke.java

```diff
@@ -1,10 +1,262 @@
+import duke.exception.AbsentArgsFlagException;
+import duke.exception.DukeException;
+import duke.exception.IllegalArgsNumException;
+import duke.exception.IllegalArgsTypeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.nio.Buffer;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
+        showLogo();
+        showWelcomeMsg();
+        readUserCmd();
+        showGoodbyeMsg();
+    }
+
+    private static void showLogo() {
+        String logo = " ____        _\n"
+                + "|  _ \\ _   _| | _____\n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        System.out.println(logo);
+    }
+
+    private static final String END_CMD = "bye";
+    private static final String DEADLINE_FLAG = "/by";
+    private static final String EVENT_FLAG = "/at";
+    private static final String[] ILLEGAL_ARGS = {"blah"};
+    private static final int MAX_TASK = 100;
```

Comment:
Naming of this constant could be improved, to better indicate if it is max size of a single task or max number of tasks in the list.

Code Quality Guideline: Name Well and Use Name to Explain